src/reference_data.rs

```diff
@@ -0,0 +1,107 @@
+use std::collections::HashMap;
+use std::env;
+
+use log::warn;
+
+use cached::proc_macro::cached;
+
+use serde_derive::Deserialize;
+
+use crate::utils::strip_http_scheme;
+
+const DEFAULT_BASE_URL: &str = "https://data.norge.no/new-reference-data";
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(non_snake_case)]
+pub struct MediaTypeCollection {
+    pub mediaTypes: Vec<MediaType>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(non_snake_case)]
+pub struct MediaType {
+    pub uri: String,
+    pub name: String,
+    pub r#type: String,
+    pub subType: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(non_snake_case)]
+pub struct FileTypeCollection {
+    pub fileTypes: Vec<FileType>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(non_snake_case)]
+pub struct FileType {
+    pub uri: String,
+    pub code: String,
+    pub mediaType: String,
+}
+
+pub fn valid_media_type(media_type: String) -> bool {
+    match get_remote_media_types() {
+        Some(media_types) => media_types.contains_key(strip_http_scheme(media_type).as_str()),
+        None => false,
+    }
+}
+
+pub fn valid_file_type(file_type: String) -> bool {
+    match get_remote_file_types() {
+        Some(file_types) => file_types.contains_key(strip_http_scheme(file_type).as_str()),
+        None => false,
+    }
+}
+
+#[cached(time = 86400)]
+pub fn get_remote_media_types() -> Option<HashMap<String, MediaType>> {
+    let base_url = env::var("REFERENCE_DATA_BASE_URL").unwrap_or(DEFAULT_BASE_URL.to_string());
+    match reqwest::blocking::get(format!("{}/iana/media-types", base_url).as_str()) {
+        Ok(resp) => match resp.json() {
+            Ok(json) => {
+                let col: MediaTypeCollection = json;
+                let mut media_types: HashMap<String, MediaType> = HashMap::new();
+                col.mediaTypes.iter().for_each(|mt| {
+                    media_types.insert(strip_http_scheme(mt.uri.clone()), mt.clone());
+                });
+
+                return Some(media_types);
+            }
+            Err(e) => {
+                warn!("Cannot get remote media-types {}", e);
+                None
+            }
+        },
+        Err(e) => {
+            warn!("Cannot get remote media-types {}", e);
+            None
+        }
+    }
+}
+
+#[cached(time = 86400)]
+pub fn get_remote_file_types() -> Option<HashMap<String, FileType>> {
+    let base_url = env::var("REFERENCE_DATA_BASE_URL").unwrap_or(DEFAULT_BASE_URL.to_string());
+    match reqwest::blocking::get(format!("{}/eu/file-types", base_url).as_str()) {
+        Ok(resp) => match resp.json() {
+            Ok(json) => {
+                let col: FileTypeCollection = json;
+                let mut file_types: HashMap<String, FileType> = HashMap::new();
+                col.fileTypes.iter().for_each(|ft| {
+                    file_types.insert(strip_http_scheme(ft.uri.clone()), ft.clone());
+                });
+
+                return Some(file_types);
```

Comment:
Det er mulig å mappe til tuples og collecte som map, da unngår man mut og insert.
Ved bruk av `into_iter` slipper man `clone()` på `ft`.
```suggestion
                return Some(
                    col.fileTypes
                        .into_iter()
                        .map(|ft| (strip_http_scheme(ft.uri.clone()), ft))
                        .collect::<HashMap<String, FileType>>(),
                );
```